mycroft/audio/audioservice.py

```diff
@@ -450,28 +450,35 @@ def _queue(self, message):
             self._play(message)
 
     def _play(self, message):
-        """
-            Handler for mycroft.audio.service.play. Starts playback of a
-            tracklist. Also  determines if the user requested a special
-            service.
+        """Start playback of a tracklist.
 
-            Args:
-                message: message bus message, not used but required
+        Also determines if the user requested a special service.
+        Handler for mycroft.audio.service.play
+
+        Args:
+            message: message bus message
         """
         with self.service_lock:
```

Comment:
As @forslund mentioned, using this context manager means that exceptions thrown within this block will correctly release the lock. Better exception handling and logging good, but it's not actually addressing the reported issue.

More likely, there needs to be a timeout parameter (with a sane default) for the play methods that are calling out to separate services/processes. If the code blocks indefinitely on an external call, there's nothing to make this lock release short of a service restart.

Reply:
the original issue was indeed vlc itself hanging forever and needing a restart, i wish it threw an exception :sweat_smile: